From 5c1061b9081a31ca89e0c01b966273b7a33f8444 Mon Sep 17 00:00:00 2001
From: David Vernet <void@manifault.com>
Date: Mon, 22 Jan 2024 12:45:28 -0600
Subject: [PATCH 103/108] scx: Replace strlcpy() with strscpy_pad()

strlcpy is removed upstream. Let's use the replacement strscpy_pad()
instead.

Signed-off-by: David Vernet <void@manifault.com>
---
 kernel/sched/ext.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/ext.c b/kernel/sched/ext.c
index 52457f65a..88a5f3476 100644
--- a/kernel/sched/ext.c
+++ b/kernel/sched/ext.c
@@ -3253,7 +3253,7 @@ static void scx_ops_disable_workfn(struct kthread_work *work)
 	}
 
 	ei->kind = kind;
-	strlcpy(ei->reason, reason, sizeof(ei->reason));
+	strscpy_pad(ei->reason, reason, sizeof(ei->reason));
 
 	/* guarantee forward progress by bypassing scx_ops */
 	scx_ops_bypass(true);
-- 
2.43.0.232.ge79552d197

